torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -473,9 +448,9 @@ void ProcessGroupAgent::markFutureWithMessage(
       futureTimeouts_.erase(rpcEndTime);
     }
   }
-  // Don't hold lock on markCompleted, as it could invoke callbacks that
-  // call agent_->send.
-  fut->markCompleted(exceptionMsg);
+  // Not holding lock on markCompleted as this could run callbacks that
+  // call agent_->send
+  fm->markCompleted(std::move(message));
```

Comment:
The code loses some efficiency here, since you started with a const Message&, and can't convert into an rval

Can we avoid this for the common non-error case?

Reply:
Could we just change the parameter to be a non-const Message&?

Reply:
oh right, this addresses the problem too.